resources/hab_build.rb

```diff
@@ -39,7 +63,10 @@
 
 action :publish do
   execute 'upload-pkg' do
-    command lazy { "#{hab_binary} pkg upload --url #{url} #{hab_studio_path}/src/results/#{artifact}" }
+    command(lazy do
+      url_opt = "--url #{url}" if url
```

Comment:
Does this need to be `new_resource.url`? Super confused on why we use `new_resource.*` in places and then in some places we call the property name directly.